src/concurrency.md

```diff
@@ -30,6 +31,36 @@ The example uses the Rayon crate, which is a data parallelism library for Rust.
 Rayon provides the `par_iter_mut()` method for any parallel iterable data type.
 It lets us write iterator-like chains that execute in parallel.
 
+[ex-rayon-parallel-sort]: #ex-rayon-parallel-sort
+<a name="ex-rayon-parallel-sort"></a>
+## Sort a vector in parallel
+
+[![rayon-badge]][rayon] [![rand-badge]][rand] [![cat-concurrency-badge]][cat-concurrency]
+
+```rust
+extern crate rand;
+extern crate rayon;
+
+use rand::Rng;
+use rayon::prelude::*;
+
+fn main() {
+    let mut vec = vec![String::new(); 100_000];
+
+    // Generating random 5 character-long Strings to be sorted later
+    vec.par_iter_mut().for_each(|p| {
+        *p = rand::weak_rng().gen_ascii_chars().take(5).collect()
+    });
+
+    // Using Rayon's parallel sort
+    vec.par_sort_unstable(); // unstable sort, like QuickSort
+}
+```
+
+The example uses the `rayon` crate, which is a data parallelism library for Rust.<br/>
+We have [`multiple options`] to sort an Iterable data type, we chose here to use [`par_sort_unstable`]
```

Comment:
This is again inspired from the Qt docs(which I think are [considerably better than the Rust docs](https://users.rust-lang.org/t/lets-talk-about-ecosystem-documentation/2791/69?u=lilianmoraru)).
I like when docs give more context and have optional links you can go to, to get more info about the topic - I don't think that it has to be strictly types only.
Here is one example from Qt: http://doc.qt.io/qt-5/qvector.html#details - pay attention that you have links to: `reentrant`(which explains the difference between thread-safe and reentrace, nothing about Qt types), `container classes`(which gives you more context about other types of containers), `constant time`(which gives you more context about the algorithmic complexity of different Qt data structures operations), `default-constructed value`(...), `implicitly shared copies`(which explains you COW in Qt).
Compared to Rust, this style of docs never left me feeling that I don't have all the info I need.

In summary: I think this is helpful to the user.